### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert: false negative with TestNG

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4503]
**Affects PMD Version:7.0.0-rc1** 

**Rule: JUnitTestsShouldIncludeAssert**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert>

**Description:**
The first code sample uses the annotations `org.testng.annotations.*`, and the second one uses `org.junit.*`. The analysis results of the two cases should be equivalent because these annotations are equivalent. However, PMD reports a warning in the second sample and no warnings in the first sample. PMD should report a warning in `Case 1` like `Case 2`.

**Code Sample demonstrating the issue:**
**Case 1**
```java
import org.testng.annotations.Test;
import org.testng.annotations.AfterClass;
public class A {
  @AfterClass
  @Test
  void test() {}  // report no warning
}
```
**Case 2**
```java
import org.junit.jupiter.api.Test;
import org.junit.AfterClass;
public class B {
    @AfterClass
    @Test
    void test() {}  // report a warning
}
```

**Expected outcome: The analysis results of the two classes should be the same and both should be reported a violation.**

PMD reports no violation in the first class, but that's wrong. That's a false negative. PMD should treat `testng` like `junit`.

**Running PMD through:** *CLI*
